tests/Controller/HolidayControllerAddTest.php

```diff
@@ -0,0 +1,293 @@
+
+        $crawler = $this->client->request('GET', '/holiday/new');
+
+        $result = $crawler->filterXPath('//body');
+        $this->assertStringContainsString('Nombre d\'heures',$result->text(),'test Conges-Mode');
+        $this->assertStringNotContainsString('Nombre de jours',$result->text(),'test Conges-Mode');
+
+        //test Conges-Mode => jours
+        $this->setParam('Conges-Mode', 'jours');
+
+        $crawler = $this->client->request('GET', '/holiday/new');
+
+        $result = $crawler->filterXPath('//body');
+        $this->assertStringContainsString('Nombre de jours',$result->text(),'test Conges-Mode');
+        $this->assertStringNotContainsString('Nombre d\'heures',$result->text(),'test Conges-Mode');
+
+        //test with Conges-Heures
+        $this->setParam('Conges-Mode', 'heures');
+        $this->setParam('Conges-Recuperations', 1);
+        $this->setParam('Conges-Heures', 1);
+
+        $crawler = $this->client->request('GET', '/holiday/new');
+
+        $button = $crawler->filterXPath('//input[@class="checkdate"]');
+        $button->click();
+
+        $result = $crawler->filterXPath('//body');
+        $this->assertStringContainsString('Heure de début',$result->text(),'test Conges-Heures');
+        $this->assertStringContainsString('Heure de fin',$result->text(),'test Conges-Heures');
+
+        //test conges anticipation, reliquat and crédit 
+
+        $result = $crawler->filterXPath('//body');
+        $this->assertStringContainsString('Reliquat : 3h48',$result->text(),'test Reliquat');
+        $this->assertStringContainsString('Crédit de congés : 3h48',$result->text(),'test crédit');
+        $this->assertStringContainsString('Solde débiteur : 0h00',$result->text(),'test solde débiteur');
+
+        //test statut with all rights
+
+        $this->setParam('Multisites-nombre', 1);
+        $this->setParam('Absences-notifications-agent-par-agent', 0);
+        $this->setParam('PlanningHebdo', 0);
+        $this->setParam('Conges-Enable', 1);
+        $this->setParam('Conges-Mode', 'Heures');
+        $this->setParam('Conges-Heures', 0);
+        $this->setParam('Conges-validation', 1);
+        $this->setParam('Conges-Validation-N2', 0);
+        $this->setParam('Conges-Recuperations', 1);
+        $this->setParam('Conges-tous', 0);
+        $this->setParam('Conges-Rappels', 0);
+        $this->setParam('Conges-Rappels-Jours', 14);
+        $this->setParam('Conges-demi-journees', 1);
+        $this->setParam('Conges-fullday-switching-time', 4);
+        $this->setParam('Conges-fullday-reference-time', '');
+        $this->setParam('Conges-planningVide', 1);
+        $this->setParam('Conges-apresValidation', 1);
+        $this->setParam('Recup-Uneparjour', 1);
+        $crawler = $this->client->request('GET', '/holiday/new');
+
+        $result = $crawler->filterXPath('//td[@id="validation-statuses"]');
+        $this->assertStringContainsString('Demandé',$result->text(),'test statut');
```

Comment:
Le test échoue
$this->assertStringContainsString('Demandé',$result->text(),'test statut');
Failed asserting that '' contains "Demandé".

Reply:
corrigé par 72e60760546677cd7cdff081dbc55ec692ee8af2